api/src/main/java/com/velocitypowered/api/proxy/server/ServerInfo.java

```diff
@@ -19,24 +21,55 @@
 public final class ServerInfo implements Comparable<ServerInfo> {
 
   private final String name;
-  private final InetSocketAddress address;
-
+  private final SocketAddress address;
+  private final AddressType addrType;
+  
   /**
    * Creates a new ServerInfo object.
    *
    * @param name the name for the server
    * @param address the address of the server to connect to
+   * @param addrType the type of the address of the server to connect to
    */
-  public ServerInfo(String name, InetSocketAddress address) {
+  public ServerInfo(String name, SocketAddress address, AddressType addrType) {
     this.name = Preconditions.checkNotNull(name, "name");
     this.address = Preconditions.checkNotNull(address, "address");
+    this.addrType = Preconditions.checkNotNull(addrType, "addrType");
+  }
+  
+  /**
+   * Creates a new ServerInfo object.
+   *
+   * @param name the name for the server
+   * @param address the address of the server to connect to
+   * @deprecated Only existing so that it won't break the api
+   */
+  @Deprecated
+  public ServerInfo(String name, InetSocketAddress address) {
+    this(name, address, AddressType.TCP);
   }
 
   public final String getName() {
     return name;
   }
-
+  
+  /**
+   * Gets the Address of the Server.
+   *
+   * {@return
+   * Returns the Address of the server to connect to
+   * or if it isn't a InetSocketAddress it returns a loopback address.
+   * }
+   */
+  @Deprecated
   public final InetSocketAddress getAddress() {
+    if (address instanceof InetSocketAddress) {
+      return (InetSocketAddress) address;
+    }
+    return InetSocketAddress.createUnresolved("127.0.0.1", 25565);
```

Comment:
This is not an acceptable default, especially because it can create unexpected duplicates in e.g. maps used by plugins.